patches/minecraft/net/minecraft/block/AbstractRailBlock.java.patch

```diff
@@ -35,12 +49,16 @@
              p_196243_2_.func_195593_d(p_196243_3_.func_177984_a(), this);
           }
  
-@@ -129,5 +_,20 @@
+@@ -129,5 +_,23 @@
        return blockstate.func_206870_a(this.func_176560_l(), flag ? RailShape.EAST_WEST : RailShape.NORTH_SOUTH);
     }
  
+-   public abstract Property<RailShape> func_176560_l();
 +   @Deprecated //Forge: Use getRailDirection(IBlockAccess, BlockPos, IBlockState, EntityMinecart) for enhanced ability
-    public abstract Property<RailShape> func_176560_l();
++   public Property<RailShape> func_176560_l()
++   {
++       return SHAPE;
++   };
```

Comment:
Why?

Reply:
Currently, any extension of AbstractRailBlock must override this function, but is not strictly required to have a RailShape Property. The code will compile, and then crash
Adding a default allows safe extension of AbstractRailBlock without having to deal with RailShape.

Reply:
No it doesn't, the subclass still needs to register this property into its block state. By making this not abstract you are even making this requirement harder to discover. Plus this is adding patch size for not much gain.

Reply:
That is unfortunate; the entire reason I started this PR was to avoid registering (an unwanted) RailShape into a subclass.

Reply:
You don't need a RailShape property if everything calls getRailDirection, that is the entire reason it exists and this method is deprecated.

Reply:
RailState.place (invoked by a vanilla rail, acting on my custom AbstractRailBlock) uses this.state = this.state.setValue(this.block.getShapeProperty(), railshape);
I don't see a way to use getRailDirection in there.

Reply:
A hook is probably needed then to provide a custom RailState implementation.